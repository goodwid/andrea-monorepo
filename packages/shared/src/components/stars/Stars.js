/* eslint-disable indent */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './Stars.scss';

import {
  FaStar as Star,
  FaRegStar as OutlinedStar,
  FaStarHalfAlt as HalfStar,
} from 'react-icons/fa';

const Stars = props => {
  const { rating } = props;

  const createStar = (starPosition) => {
    const Icon = rating >= starPosition       ? Star : 
                 rating > starPosition - 0.6  ? HalfStar :
                                                OutlinedStar;
    return (<Icon className="icon" />);
  };

  return (
    <span className={styles.Stars}>
      { createStar(1) }
      { createStar(2) }
      { createStar(3) }
      { createStar(4) }
      { createStar(5) }  
    </span>
  );
};

Stars.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default Stars;
