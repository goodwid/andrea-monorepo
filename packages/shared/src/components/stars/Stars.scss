@import 'shared/styles/variables.scss';

:local(.Stars) {
    .icon {
        margin-bottom: 1rem;
        color: $gray;
    }
}