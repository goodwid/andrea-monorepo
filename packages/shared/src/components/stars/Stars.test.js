import React from 'react';
import { shallow } from 'enzyme';
import Stars from './Stars';

describe('Stars page', () => {
  it('matches the snapshot', () => {
    const rating = 4;

    const wrapper = shallow(
      <Stars rating={rating}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
