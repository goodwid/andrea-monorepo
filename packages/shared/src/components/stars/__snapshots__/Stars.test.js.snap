// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars page matches the snapshot 1`] = `
<span
  className="Stars"
>
  <FaStar
    className="icon"
  />
  <FaStar
    className="icon"
  />
  <FaStar
    className="icon"
  />
  <FaStar
    className="icon"
  />
  <FaRegStar
    className="icon"
  />
</span>
`;
